src/ert/ensemble_evaluator/evaluator.py

```diff
@@ -305,10 +305,17 @@ async def _server(self) -> None:
                 self._router_socket.curve_publickey = self._config.server_public_key
                 self._router_socket.curve_server = True
 
-            if self._config.router_port:
-                self._router_socket.bind(f"tcp://*:{self._config.router_port}")
+            if self._config.use_ipc_protocol:
+                self._router_socket.bind(self._config.get_uri())
+            if self._config.port_range[0] == self._config.port_range[1]:
+                self._router_socket.bind(f"tcp://*:{self._config.port_range[0]}")
+                self._config.router_port = self._config.port_range[0]
             else:
-                self._router_socket.bind(self._config.url)
+                self._config.router_port = self._router_socket.bind_to_random_port(
+                    "tcp://*",
+                    min_port=self._config.port_range[0],
+                    max_port=self._config.port_range[1],
```

Comment:
let's also use by default it tries to bind 100 times. `max_tries: int = 100` maybe we can lower it?

Reply:
it runs quick even with 100 tries. test_evaluator_raises_on_start_with_address_in_use passes in 0.14s on my computer.